### PR TITLE
Fix React + MongoDB

### LINKED ISSUE #14100: React + MongoDB + Protractor : e2e broken
See:
- https://github.com/hipster-labs/jhipster-daily-builds/actions/workflows/react-maven-nosql.yaml
- https://github.com/hipster-labs/jhipster-daily-builds/actions/workflows/react-gradle-nosql.yaml

### PR DESCRIPTION
**Pull request by mraible**
Fixes #14100.

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by mshima** [comment]
@mraible can you try `autoGenerate` instead of `jpaGeneratedValue`.

**Comment by mraible** [comment]
@gzsombor I had to revert your change for embedded documents to get the build to work. IMO, the ID field should be hidden when you're creating a new record. No reason to show something the user can't change.

Daily builds to see if this works:

* https://github.com/hipster-labs/jhipster-daily-builds/actions/runs/615598222
* https://github.com/hipster-labs/jhipster-daily-builds/actions/runs/615598835

**Comment by mraible** [comment]
The `gradle-react-cassandra` job fails in the daily builds. I ran it locally and everything passes, so I'm not too concerned. I'll try reverting my changes now and using @mshima's suggestion.

**Comment by mraible** [comment]
@mshima I tried changing it to `autoGenerate` in this line:

```
const readonlyIfGeneratedId = ((field.id && (field.autoGenerate || databaseType !== 'sql')) ? ' readOnly' : '');
```

The ID field is still displayed and e2e tests fail.

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs [hunk 1]
```diff
@@ -227,11 +227,17 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
       <Col md="8">
         { loading ? <p>Loading...</p> :
         <AvForm model={isNew ? {} : <%= entityInstance %>Entity} onSubmit={saveEntity} >
-          <%_ for (field of fields) {
+          { !isNew ?
+            <AvGroup>
+              <Label for="<%= entityFileName %>-id"><Translate contentKey="global.field.id">ID</Translate></Label>
+              <AvInput id="<%= entityFileName %>-id" type="text" className="form-control" name="id" required readOnly/>
```
Comment (mshima) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs:233]: `name="id"` will break custom ids support.
Should use `${ primaryKey.name }`.

Comment (mraible) [reply generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs:233]: Like this?

```jsx
<AvGroup>
  <Label for="<%= entityFileName %>-<%= primaryKey.name %>"><Translate contentKey="global.field.<%= primaryKey.name %>">ID</Translate></Label>
  <AvInput id="<%= entityFileName %>-<%= primaryKey.name %>" type="text" className="form-control" name="<%= primaryKey.name %>" required readOnly/>
</AvGroup>
```

Comment (mshima) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs:233]: My bad, I took deeper look and the fix should be something like:
```suggestion
              <Label for="<%= entityFileName %>-<%= primaryKey.fields[0].fieldName %>"><Translate contentKey="<%= primaryKey.fields[0].fieldTranslationKey %>"><%= primaryKey.fields[0].fieldNameHumanized %></Translate></Label>
              <AvInput id="<%= entityFileName %>-<%= primaryKey.fields[0].fieldName %>" type="text" className="form-control" name="<%= primaryKey.fields[0].fieldName %>" required readOnly/>
```
`fieldTranslationKey` and `fieldNameHumanized` are not available directly in the primaryKey.

Comment (mshima) [reply generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs:233]: I've put another suggestion below.

**Comment by deepu105** [comment]
I was gonna look into this. Seems like you beat me to it :)

**Comment by mraible** [comment]
If these two daily builds pass, I'm going to merge this PR:

* Maven + React: https://github.com/hipster-labs/jhipster-daily-builds/actions/runs/618550495
* Gradle + React: https://github.com/hipster-labs/jhipster-daily-builds/actions/runs/618550987

**Pull request merged**